src/routes/auth_routes.test.ts

```diff
@@ -0,0 +1,1540 @@
+      { id: 1, deviceLoginLimit: limit },
+    ]);
+  },
+};
+
+const setupAuthenticatedRequest = (user: MockUser = createMockUser()) => {
+  return (_req: Request, _res: Response, next: NextFunction) => {
+    const req = _req as RequestWithUser;
+    req.user = user;
+    req.isAuthenticated = jest.fn().mockReturnValue(true);
+    req.logout = jest.fn();
+    req.sessionID = 'mock-session-id';
+    next();
+  };
+};
+
+const setupUnauthenticatedRequest = () => {
+  return (_req: Request, res: Response) => {
+    res.status(401).send('Unauthorized');
+  };
+};
+
+const setupPassportAuthenticate = (
+  behavior: 'success' | 'failure' | 'custom',
+  customCallback?: (req: Request, res: Response, next?: NextFunction) => void,
+) => {
+  const mockAuthenticate = jest.fn().mockImplementation(() => {
+    return (req: Request, res: Response, next?: NextFunction) => {
+      switch (behavior) {
+        case 'success':
+          res.redirect('/');
+          break;
+        case 'failure':
+          res.redirect('/login/fail');
+          break;
+        case 'custom':
+          if (customCallback) customCallback(req, res, next);
+          break;
+      }
+    };
+  });
+
+  passport.authenticate = mockAuthenticate;
+  return mockAuthenticate;
+};
+
+// ===== TEST SUITE =====
+describe('auth_routes', () => {
+  let app: Express;
+  let mockAuthFuncs: Record<
+    string,
+    { logout: jest.Mock; getMetaData: jest.Mock }
+  >;
+  let mockEnsureAuthenticated: jest.Mock;
+  let mockLogIn: jest.Mock;
+  let mockLog: jest.Mock;
+
+  beforeEach(() => {
+    // Reset all mocks and set up successful defaults
+    jest.clearAllMocks();
+    setupSuccessfulMocks();
+
+    // Create fresh Express app for each test
+    app = express();
+    app.use(express.json());
+    app.use(express.urlencoded({ extended: true }));
+
+    // Create mock functions
+    mockAuthFuncs = {
+      'example.com': {
+        logout: jest.fn((req: Request, res: Response) => {
+          res.redirect('/logout/callback');
+        }),
+        getMetaData: jest.fn().mockReturnValue('<xml>metadata</xml>'),
+      },
+    };
+    mockEnsureAuthenticated = jest.fn();
+    mockLogIn = jest.fn();
+    mockLog = jest.fn();
+
+    // Mock global dependencies
+    global.connection = {
+      query: jest.fn(
+        (
+          query: string,
+          params: unknown,
+          callback?: (err: Error | null, results: unknown[]) => void,
+        ) => {
+          if (typeof params === 'function') {
+            const cb = params as (
+              err: Error | null,
+              results: unknown[],
+            ) => void;
+            cb(null, [createMockIDP()]);
+          } else if (callback) {
+            callback(null, [createMockIDP()]);
+          }
+        },
+      ),
+    };
+
+    // Mock process.env
+    process.env.LOGIN_TEST_EMAIL = 'test@example.com';
+    process.env.LOGIN_TEST_CODE = 'TEST123';
+    process.env.IS_DEV = 'true';
```

Comment:
Is there a reason for setting IS_DEV? I would expect the tests to ensure that production will function correctly